### PR TITLE
Adding hybrid implementation of 'cleanReSyncGhosts' API

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

**Comment by wmathurin** [comment]
👍

File: gen/plugins/com.salesforce/com.salesforce.util.push.js [hunk 1]
```diff
@@ -31,61 +31,45 @@ var SALESFORCE_MOBILE_SDK_VERSION = "4.2.0";
...
-            notificationHandler(message);
-        }
-    };
-    
-    var registrationSuccess = function(result) {
-        console.log("Registration successful " + JSON.stringify(result));
-    };
+        var push = PushNotification.init({
+                "android": {
+                    "senderID": bootconfig.androidPushNotificationClientId
```
Comment (kchitalia) [inline gen/plugins/com.salesforce/com.salesforce.util.push.js]: + :100: We are now compatible with latest version of push plugin. Thank you!!!

File: gen/plugins/com.salesforce/com.salesforce.plugin.smartsync.js [hunk 2]
```diff
@@ -111,5 +118,6 @@ module.exports = {
     syncDown: syncDown,
     syncUp: syncUp,
     getSyncStatus: getSyncStatus,
-    reSync: reSync
+    reSync: reSync,
+    cleanReSyncGhosts: cleanReSyncGhosts
```
Comment (kchitalia) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartsync.js]: Need to expose  resetSyncManager here?

Comment (bhariharan) [inline gen/plugins/com.salesforce/com.salesforce.plugin.smartsync.js]: Yes - need it for tests and in general it's a public API.

**Pull request merged**